File: IdeaProjects/Resources/ReadingFromCommandLineExamples/CommandLineExample.java
import java.io.*;
import java.util.ArrayList;

public class CommandLineExample {
    public static void main(String[] args){

        String fileName = args.length == 1 ? args[0] : "input.txt";

        ArrayList<String> linesList = null;

        try{ linesList = readFileToList(fileName); }
        catch(IOException e){
            System.out.println(e.getMessage() +
                    "\nTry placing the input file in this directory" +
                    "\n" + System.getProperty("user.dir"));
            System.exit(1);
        }

        linesList.forEach((line) -> System.out.println(line));

    }

    private static ArrayList<String> readFileToList(String fileName) throws IOException {

        File input = new File(fileName);
        BufferedReader reader = new BufferedReader(new FileReader(fileName));

        String line = reader.readLine();

        ArrayList<String> linesList = new ArrayList<String>();

        while(line != null){
            linesList.add(line);
            line = reader.readLine();

        }

        return linesList;
    }
}

File: IdeaProjects/Resources/ReadingFromCommandLineExamples/CommandLineExample.kt
import java.io.File
import java.io.FileNotFoundException

fun main(args : Array<String>){

    var fileName = if(args.size == 1) args[0] else "input.txt"

    lateinit var lineList : List<String>

    try{ lineList = readFileToArray(fileName) }
    catch(e : FileNotFoundException){
        println(e.message +
                "\nTry placing the input file in the following directory: "
                        + System.getProperty("user.dir"))
        System.exit(0)
    }

    print(lineList)
}

fun readFileToArray(fileName : String) : List<String> = File(fileName).readLines()